src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java

```diff
@@ -502,6 +505,87 @@ public OrganisationEntityResponseV2 retrieveOrganisationForV2Api(
     }
 
+    @Override
+    public MultipleOrganisationsResponse retrieveOrganisationsByProfileIds(List<String> organisationProfileIds,
+                                                                           UUID searchAfter) {
+        List<String> orgTypes = convertProfileIdsToOrgTypes(organisationProfileIds);
+
+        boolean orgIdFilterProvided = !orgTypes.isEmpty();
+        boolean searchAfterProvided = searchAfter != null;
+
+        List<Organisation> organisations = new ArrayList<>();
+
+        if (orgIdFilterProvided && searchAfterProvided) {
+            organisations = organisationRepository.findByOrgTypeInAndIdGreaterThan(orgTypes, searchAfter);
+        }
+
+        if (orgIdFilterProvided && !searchAfterProvided) {
+            organisations = organisationRepository.findByOrgTypeIn(orgTypes);
+        }
+
+        if (!orgIdFilterProvided && searchAfterProvided) {
+            organisations = organisationRepository.findByIdGreaterThan(searchAfter);
+        }
+
+        if (!orgIdFilterProvided && !searchAfterProvided) {
+            organisations = organisationRepository.findAll();
+        }
+
+        return new MultipleOrganisationsResponse(organisations, false);
+    }
+
+    @Override
+    public MultipleOrganisationsResponse retrieveOrganisationsByProfileIdsWithPageable(
+            List<String> organisationProfileIds,
+            Integer pageSize,
+            UUID searchAfter) {
+        List<String> orgTypes = convertProfileIdsToOrgTypes(organisationProfileIds);
+
+        Pageable pageableObject = createPageableObject(0, pageSize, Sort.by(Sort.DEFAULT_DIRECTION, "id"));
+        boolean orgIdFilterProvided = organisationProfileIds != null && !organisationProfileIds.isEmpty();
+        boolean searchAfterProvided = searchAfter != null;
+
+        Page<Organisation> orgs = null;
+
+        if (orgIdFilterProvided && searchAfterProvided) {
+            orgs = organisationRepository
+                    .findByOrgTypeInAndIdGreaterThan(orgTypes, searchAfter, pageableObject);
+        }
+
+        if (orgIdFilterProvided && !searchAfterProvided) {
+            orgs = organisationRepository.findByOrgTypeIn(orgTypes, pageableObject);
+        }
+
+        if (!orgIdFilterProvided && searchAfterProvided) {
+            orgs = organisationRepository.findByIdGreaterThan(searchAfter, pageableObject);
+        }
+
+        if (orgs == null) {
+            return new MultipleOrganisationsResponse(new ArrayList<>(), false);
+        }
+        return new MultipleOrganisationsResponse(orgs.getContent(), !orgs.isLast());
+    }
+
+    private static List<String> convertProfileIdsToOrgTypes(List<String> organisationProfileIds) {
+        List<String> orgTypes = Optional.ofNullable(organisationProfileIds)
+                .orElse(Collections.emptyList())
+                .stream()
+                .map(profileId -> {
+                    switch (profileId) {
+                        case "SOLICITOR_PROFILE":
```

Comment:
why these are hardcoded here, can we create enum for these

Reply:
Create the enum in the util folder